components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go

```diff
@@ -23,6 +29,9 @@ type FunctionSpec struct {
 	// Deps defines the dependencies for a function
 	Deps string `json:"deps,omitempty"`
 
+	// +optional
+	Runtime Runtime `json:"runtime, omitempty"`
```

Comment:
i'm not sure whether that space after `runtime,` is correct, it could introduce some nasty bug